src/trues.jl

```diff
@@ -43,3 +43,11 @@ function Base.getindex(x::AbstractArray{T,N}, mask::Trues{N, NTuple{N,Base.OneTo
     end
     return x[trues(size(x))] # else revert to usual getindex method
 end
+
+# https://github.com/JuliaArrays/FillArrays.jl/issues/148
+function Base.getindex(
+    a::AbstractFill{T, 1, Axes} where Axes,
+    b::Trues{1, Tuple{Base.OneTo{Int}}}) where T
+    @boundscheck length(a) == length(b) || throw(BoundsError(a, b))
+    return Fill(getindex_value(a), length(a))
```

Comment:
Need to check ` axes ` match

Reply:
Short answer:
Because `b` here is set to be `Tuple{Base.OneTo{Int}}}`, instead of adding check and a test, I have simply specialized the method so that it applies only to `a` with those same axes.  That will suffice for my use cases, and what to do more generally seems unclear.

You can ignore the rest of this comment if you are satisfied with that solution.
I am writing it both as notes for myself and in case it helps someone else generalize the method in the future if there is a need.

I am unsure that the `axes` must match when doing `getindex` with a 1D Bool array like we are here.

The following 1D logical indexing on RHS works fine in Base:
```julia
ones(2,3)[trues(2,3)] == ones(2,3)[trues(6)]
```

And similar 1D logical indexing works fine in `FillArrays` already:
```julia
Ones((3:5, 2:3))[trues(6)]
```

The axes do not even have the same dimensions in those two examples, so clearly the axes do not match, yet it works as expected.
In general when one does `getindex` with a 1D Bool vector, the axes become moot,
because of arbitrary sampling, e.g.,
`(2:4)[[true, false, true]]`
or even
`Ones(3)[[true, false, true]]`

On the other hand, this throws an error in FA currently:
`Ones((2:4,))[[true,false,true]]`

It is unclear to me from the docs whether throwing an error is the right thing to do or not.
https://docs.julialang.org/en/v1/devdocs/offset-arrays/
Resolving that (if needed) would be a separate issue from this PR.